modules/distribution-zones/src/main/java/org/apache/ignite/internal/distributionzones/DistributionZoneManager.java

```diff
@@ -1153,10 +1277,24 @@ synchronized void rescheduleScaleDown(long delay, Runnable runnable) {
          * Cancels task for scale up and scale down.
          */
         synchronized void stopTimers() {
+            stopScaleUp();
+
+            stopScaleDown();
+        }
+
+        /**
+         * Cancels task for scale up.
+         */
+        synchronized void stopScaleUp() {
```

Comment:
Why do we need synchronized here? Not a big deal though. However, as mentioned bellow in case of using synchronized instead of volatile you should also read tasks from within synchronized block.